src/tests/hash_functions/test_64_bit_hash_performance.f90

```diff
@@ -0,0 +1,161 @@
+program test_64_bit_hash_performance
+!! Program to compare the relative performance of different 64 bit hash
+!! functions
```

Comment:
Is this also testing correctness (not just performance)? I'm not an expert on hashing, but it isn't clear to me whether the code would fail if we introduced a bug into the hashing functions (obviously, we would want that to happen).

Reply:
It is unlikely to fail on these tests, they are performance only, but I do have codes outside the standard library for checking correctness.